OpenRA.Mods.Common/ServerTraits/LobbyCommands.cs

```diff
@@ -83,682 +83,736 @@ public bool InterpretCommand(S server, Connection conn, Session.Client client, s
 
 			var dict = new Dictionary<string, Func<string, bool>>
 			{
-				{ "state",
-					s =>
-					{
-						var state = Session.ClientState.Invalid;
-						if (!Enum<Session.ClientState>.TryParse(s, false, out state))
-						{
-							server.SendOrderTo(conn, "Message", "Malformed state command");
-							return true;
-						}
+				{ "state", State(server, conn, client) },
+				{ "startgame", StartGame(server, conn, client) },
```

Comment:
These all take the same arguments, so we can simplify the definitions and remove a level of abstraction by referring to the static methods directly:
```csharp
public bool InterpretCommand(S server, Connection conn, Session.Client client, string cmd)
{
	if (server == null || conn == null || client == null || !ValidateCommand(server, conn, client, cmd))
		return false;

	var dict = new Dictionary<string, Func<S, Connection, Session.Client, string, bool>>
	{
		{ "state", State },
		// Other command definitions
	};

	var cmdName = cmd.Split(' ').First();
	var cmdValue = cmd.Split(' ').Skip(1).JoinWith(" ");

	Func<S, Connection, Session.Client, string, bool> a;
	if (!dict.TryGetValue(cmdName, out a))
		return false;

	return a(server, conn, client, cmdValue);
}

static bool State(S server, Connection conn, Session.Client client, string s)
{
	var state = Session.ClientState.Invalid;
	// .. the rest of the body of State, removing the s => { } wrapper
	return true;
}

// Other command definitions
```

Reply:
The dict definition could then be pulled out to a field on the object, which avoids unnecessary allocations each time the `InterpretCommand` method is called.